crates/ef-testing/src/evm_sequencer/evm_state/v0.rs

```diff
@@ -96,189 +96,189 @@ impl Evm for KakarotSequencer {
-        self.state_mut().set_storage_at(
-            kakarot_address,
-            get_storage_var_address(KAKAROT_EVM_TO_STARKNET_ADDRESS, &[account.evm_address]),
-            *starknet_address.0.key(),
-        )?;
-        Ok(())
-    }
+            storage.append(&mut vec![
+                starknet_storage!(ACCOUNT_IMPLEMENTATION, self.environment.eoa_class_hash.0), // both EOA and CA CH are the same (for now)
+                starknet_storage!(
+                    ACCOUNT_CAIRO1_HELPERS_CLASS,
+                    self.environment.cairo1_helpers_class_hash.0
+                ), // both EOA and CA CH are the same (for now)
+                starknet_storage!(OWNABLE_OWNER, *self.environment.kakarot_address.0.key()),
+            ]);
+
+            // Write all the storage vars to the sequencer state.
+            for (k, v) in storage {
+                self.state_mut().set_storage_at(starknet_address, k, v)?;
+            }
+
+            let class_hash = self.environment.contract_account_class_hash;
+            // Set up the contract class hash
+            self.state_mut()
+                .set_class_hash_at(starknet_address, class_hash)?;
 
-    /// Funds an EOA or contract account. Also gives allowance to the Kakarot contract.
-    fn fund(&mut self, evm_address: &Address, balance: U256) -> StateResult<()> {
-        let starknet_address = self.compute_starknet_address(evm_address)?;
-        let balance_values = split_u256(balance);
-        let mut storage = vec![];
-
-        // Initialize the balance storage var.
-        let balance_low_key = get_fee_token_var_address(starknet_address);
-        let balance_high_key = next_storage_key(&balance_low_key)?;
-        storage.append(&mut vec![
-            (balance_low_key, StarkFelt::from(balance_values[0])),
-            (balance_high_key, StarkFelt::from(balance_values[1])),
-        ]);
-
-        // Initialize the allowance storage var.
-        let allowance_key_low = get_storage_var_address(
-            "ERC20_allowances",
-            &[
+            // Add the address to the Kakarot evm to starknet mapping
+            let kakarot_address = self.environment.kakarot_address;
+            self.state_mut().set_storage_at(
+                kakarot_address,
+                get_storage_var_address(KAKAROT_EVM_TO_STARKNET_ADDRESS, &[account.evm_address]),
                 *starknet_address.0.key(),
-                *self.environment.kakarot_address.0.key(),
-            ],
-        );
-        let allowance_key_high = next_storage_key(&allowance_key_low)?;
-        storage.append(&mut vec![
-            (allowance_key_low, StarkFelt::from(u128::MAX)),
-            (allowance_key_high, StarkFelt::from(u128::MAX)),
-        ]);
-
-        // Write all the storage vars to the sequencer state.
-        for (k, v) in storage {
-            self.state_mut()
-                .set_storage_at(*ETH_FEE_TOKEN_ADDRESS, k, v)?;
+            )?;
+            Ok(())
```

Comment:
you have an issue here, the Ok(()) should be outside the if. otherwise you don't have the correct return types